prowler/exceptions/exceptions.py

```diff
@@ -0,0 +1,68 @@
+class ProwlerException(Exception):
+    """Base exception for all Prowler SDK errors."""
+
+    ERROR_CODES = {
+        (1901, "ProviderConnectionError"): {
+            "message": "Provider connection error",
+            "remediation": "Check your network connection and ensure the service is reachable.",
+            "file": "{file}",
+            "provider": "{provider}",
+        },
+        (1902, "ProviderAuthenticationError"): {
+            "message": "Provider authentication failed",
+            "remediation": "Verify your credentials and ensure they have the necessary permissions.",
+            "file": "{file}",
+            "provider": "{provider}",
+        },
+        (1903, "ProviderTimeoutError"): {
+            "message": "Request to provider timed out",
+            "remediation": "Consider increasing the timeout setting or check the service status.",
+            "file": "{file}",
+            "provider": "{provider}",
+        },
+        (1905, "FileExistsError"): {
+            "message": "File could not be updated, it already exists",
+            "remediation": "Provide a different file or set overwrite=True to overwrite the existing file.",
+            "file": "{file}",
+            "provider": "{provider}",
+        },
+    }
+
+    def __init__(
```

Comment:
Please document de `init` and the class with more details.

Reply:
Done